src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy

```diff
@@ -1594,6 +1708,230 @@ class PriceFloorsFetchingSpec extends PriceFloorsBaseSpec {
+
+        when: "PBS processes auction request"
+        def bidResponse = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS shouldn't log warning or errors"
+        assert !bidResponse.ext?.warnings
+        assert !bidResponse.ext?.errors
+
+        and: "PBS shouldn't log a errors"
+        def message = 'Price floor rules data must be present'
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, PRICE_FLOORS_ERROR_LOG(bidRequest, FETCHING_DISABLED_ERROR, message))
+        assert !floorsLogs.size()
+
+        and: "Alerts.general metrics shouldn't be populated"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert !metrics[ALERT_GENERAL]
+    }
+
+    def "PBS shouldn't emit error in log and response when data is invalid and floors fetching enabled for account"() {
+        given: "Default BidRequest with empty floors.data"
+        def bidRequest = bidRequestWithFloors.tap {
+            ext.prebid.floors.enabled = requestFloors
+            ext.prebid.floors.data = null
+        }
+
+        and: "Account with enabled fetching"
+        def account = getAccountWithEnabledFetch(bidRequest.accountId).tap {
+            config.auction.priceFloors.enabled = true
+            config.auction.priceFloors.fetch.enabled = true
+        }
+        accountDao.save(account)
+
+        and: "Default bid response"
+        def response = BidResponse.getDefaultBidResponse(bidRequest)
+        bidder.setResponse(bidRequest.id, response)
+
+        and: "Flush metrics"
+        flushMetrics(floorsPbsService)
+
+        when: "PBS processes auction request"
+        def bidResponse = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS shouldn't log warning or errors"
+        assert !bidResponse.ext?.warnings
+        assert !bidResponse.ext?.errors
+
+        and: "PBS shouldn't log a errors"
+        def message = 'Price floor rules data must be present'
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, PRICE_FLOORS_ERROR_LOG(bidRequest, FETCHING_DISABLED_ERROR, message))
+        assert !floorsLogs.size()
+
+        and: "PBS request on response object status should be in progress"
+        assert getRequests(bidResponse)[GENERIC.value]?.first?.ext?.prebid?.floors?.fetchStatus == INPROGRESS
+
+        and: "PBS bidderRequest status should be in progress"
+        def bidderRequest = bidder.getBidderRequests(bidRequest.id)
+        assert bidderRequest.ext.prebid.floors.fetchStatus == [INPROGRESS]
+
+        and: "Alerts.general metrics shouldn't be populated"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert !metrics[ALERT_GENERAL]
+
+        where:
+        requestFloors << [null, true]
+    }
+
+    def "PBS shouldn't emit error in log and response when data is invalid and floors disabled for request"() {
```

Comment:
This one looks like the same as `PBS shouldn't emit error in log and response when floor data is empty and floors fetching disabled for account and disabled for request`

Reply:
Or it can be merged